process/common_test.go

```diff
@@ -1247,42 +1247,8 @@ func TestGetShardHeaderFromStorageWithNonceShouldWork(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, hash, headerHash)
 	assert.Equal(t, hdr, header)
-}
 
-func TestGetHeaderFromStorageWithNonceShouldWorkForShard(t *testing.T) {
-	nonce := uint64(1)
-	shardId := uint32(0)
-	hash := []byte("X")
-	nonceToByte := []byte("1")
-	hdr := &block.Header{Nonce: nonce}
-	marshalizer := &mock.MarshalizerMock{}
-	marshHdr, _ := marshalizer.Marshal(hdr)
-	storageService := &mock.ChainStorerMock{
-		GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
-			return &mock.StorerStub{
-				GetCalled: func(key []byte) ([]byte, error) {
-					if bytes.Equal(key, nonceToByte) {
-						return hash, nil
-					}
-					if bytes.Equal(key, hash) {
-						return marshHdr, nil
-					}
-					return nil, errors.New("error")
-				},
-			}
-		},
-	}
-	uint64Converter := &mock.Uint64ByteSliceConverterMock{
-		ToByteSliceCalled: func(n uint64) []byte {
-			if n == nonce {
-				return nonceToByte
-			}
-
-			return nil
-		},
-	}
-
-	header, headerHash, err := process.GetHeaderFromStorageWithNonce(
+	header, headerHash, err = process.GetShardHeaderFromStorageWithNonce(
```

Comment:
Here there are two functions tested, I would leave two separate tests, but for the storageService initialization (which is the duplicated code) I would move to a initDefaultStorageService funciton and call that in both tests.

Reply:
done.